pkg/agent/core/ngt/service/kvs/kvs_test.go

```diff
@@ -219,62 +299,141 @@ func Test_bidi_GetInverse(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           val: 0,
-		       },
-		       fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           val: 0,
-		           },
-		           fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "return key and true when there is a key for the value",
+				args: args{
+					val: val,
+				},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  key,
+					want1: true,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key         = "45637ec4-c85f-11ea-87d0"
+				val1 uint32 = 14438
+			)
+			var val2 uint32 = 10000
+
+			return test{
+				name: "return false when there is a no key for the value",
+				args: args{
+					val: val2,
+				},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val1)
+				},
+				want: want{
+					want:  "",
+					want1: false,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name:   "return false when there is a no key for the value and the val is 0",
+				args:   args{},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  "",
+					want1: false,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = ""
+				val uint32 = 0
+			)
+
+			return test{
+				name:   "return true when set the default value and there is a key for the value and the val is 0",
+				args:   args{},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  "",
+					want1: true,
+				},
+			}
```

Comment:
sorry again but can you add new test cases when the same key is set twice and see if it will override the value that is set?